src/pretix/control/forms/organizer_forms/organizer_form.py

```diff
@@ -22,3 +27,153 @@ def clean_slug(self):
+        max_length=255,
+        widget=forms.TextInput(attrs={"placeholder": ""}),
+    )
+
+    zip_code = forms.CharField(
+        label=_("Zip Code"),
+        help_text=_("Enter your postal code."),
+        required=True,
+        max_length=255,
+        widget=forms.TextInput(attrs={"placeholder": ""}),
+    )
+
+    city = forms.CharField(
+        label=_("City"),
+        help_text=_("Enter your city."),
+        required=True,
+        max_length=255,
+        widget=forms.TextInput(attrs={"placeholder": ""}),
+    )
+
+    country = forms.ChoiceField(
+        label=_("Country"),
+        help_text=_("Select your country."),
+        required=True,
+        choices=CachedCountries(),
+        initial="US",
+    )
+
+    preferred_language = forms.ChoiceField(
+        label=_("Preferred Language for Correspondence"),
+        help_text=_("Select your preferred language for all communication."),
+        required=True,
+    )
+
+    tax_id = forms.CharField(
+        label=_("Tax ID (e.g., VAT, GST)"),
+        help_text=_(
+            "If you are located in the EU, please provide your VAT ID. "
+            "Without this, we will need to charge VAT on our services and will not be able to issue reverse charge invoices."
+        ),
+        max_length=255,
+        widget=forms.TextInput(attrs={"placeholder": ""}),
+        required=False,
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.organizer = kwargs.pop("organizer", None)
+        selected_languages = [
+            (code, name)
+            for code, name in settings.LANGUAGES
+            if code in self.organizer.settings.locales
+        ]
+        self.base_fields["preferred_language"].choices = selected_languages
+        self.base_fields["preferred_language"].initial = self.organizer.settings.locale
+        super().__init__(*args, **kwargs)
+        self.set_initial_data()
+
+    def set_initial_data(self):
+        billing_settings = OrganizerBillingModel.objects.filter(
+            organizer_id=self.organizer.id
+        ).first()
+
+        if billing_settings:
+            for field in self.Meta.fields:
+                self.initial[field] = getattr(billing_settings, field, "")
+
+    def save(self, commit=True):
+        instance = super().save(commit=False)
+        instance.organizer_id = self.organizer.id
+        billing_settings = OrganizerBillingModel.objects.filter(
+            organizer_id=self.organizer.id
+        ).first()
+
+        if billing_settings:
+            for field in self.Meta.fields:
+                setattr(billing_settings, field, self.cleaned_data[field])
+            if commit:
+                update_customer_info(
+                    billing_settings.stripe_customer_id,
+                    email=self.cleaned_data.get("primary_contact_email"),
+                    name=self.cleaned_data.get("primary_contact_name"),
+                )
+                billing_settings.save()
+            return billing_settings
```

Comment:
Why this method suddenly returns a different object than what is declared in `Meta`? This is a hidden magic which will cause confusion and bugs in the future.